### PR TITLE
scroll fields from added hetero-list entry into viewport

### PR DESCRIPTION
**Pull request by mawinter69**
The current logic in ensureVisible seems to have no effect, so it is changed in the following way:
1. When the element is bigger then the current viewport height, scroll so that the element is at the top.
2. When the elements is smaller than the viewport height and the bottom is outside, scroll so that the bottom is at the lower end of the viewport.
3. Otherwise no scrolling is needed as the element is already completely inside the viewport.

This also replaces usage of YUI with plain javascript in the ensureVisible method

https://github.com/user-attachments/assets/1a21e4d1-6416-4fa7-89b6-996ab9a0d737

https://github.com/user-attachments/assets/74ac70fc-d913-4657-a348-a5ab8ad3821d

https://github.com/user-attachments/assets/07312ff3-e362-4f69-b752-123ba42ffa63

### Testing done
Manual testing

### Proposed changelog entries

- Scroll fields from added hetero-list entry into viewport.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Comment by timja** [comment]
@mawinter69 you able to run an ATH build please? It can be sensitive to this sort of thing

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (zbynek) [inline war/src/main/webapp/scripts/hudson-behavior.js:2245]: The current scroll position is read from either `body` or `documentElement`, but the new value is only applied to `documentElement`. AFAIK recent browser versions use `documentElement` for the global scroll, so the part with `body` can be omitted?

**Comment by mawinter69** [comment]
I've run ath here: https://github.com/jenkinsci/acceptance-test-harness/pull/1628 without problems

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (mawinter69) [reply war/src/main/webapp/scripts/hudson-behavior.js]: I copied that code from yahoo. Removed the part with body.

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by zbynek — APPROVED** [review]

**Pull request merged**
